File: FRONT/personaje-frontend/src/app/personaje.ts
export class InfoDto {
    count: number;
    pages: number;
    next: string | null;
    prev: string | null;
  }

  export class LocationDto {
    name: string;
    url: string;
  }

  export class PersonajeResponseDto {
    id: number;
    name: string;
    status: string;
    species: string;
    type: string;
    gender: string;
    origin: LocationDto;
    location: LocationDto;
    image: string;
    episode: string[];
    url: string;
    created: string;
  }

  export class ListaPersonajes {
    info:{count: number;
    pages: number;
    next: string | null;
    prev: string | null;
    };
    results: Character[];
  }

  export class Character {
    id: number;
    name: string;
    status: string;
    species: string;
    gender: string;
    origin: { name: string; url: string };
    location: { name: string; url: string };
    image: string;
    episode: string[];
    url: string;
    created: string;
  }

File: personaje-backend/src/main/java/com/example/cliente/model/Personaje.java
package com.example.cliente.model;

import com.example.cliente.dto.response.LocationDto;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;
@Getter // genera todos los getters
@Setter // genera todos los setters
@AllArgsConstructor
@NoArgsConstructor
public class Personaje {

    private Long id;
    private String name;
    private String status;
    private String species;
    private String type;
    private String gender;
    private LocationDto origin;
    private LocationDto location;
    private String image;
    private List<String> episode;
    private String url;
    private String created;

}
